test/unit/bucket/test_bucket.py

```diff
@@ -661,7 +665,27 @@ def test_matching_single_question_mark_named_file(self):
         ]
         actual = [
             (info.file_name, info.size, info.action, folder)
-            for (info, folder) in self.bucket_ls('b/?.txt', with_wildcard=True)
+            for (info, folder) in self.bucket_ls('b/?.txt', recursive=True, with_wildcard=True)
+        ]
+        self.assertEqual(expected, actual)
+
+    @pytest.mark.apiver(from_ver=2)
```

Comment:
it doesn't work on v1?

Reply:
`Bucket.ls` was not forwarding `kwargs`. Fixed.